src/tests/multihost/sssd/testlib/common/custom_log.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+import os
+from collections import defaultdict
+
+import pytest
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_configure(config: pytest.Config) -> None:
+    p = CustomLogPlugin(config)
+    config.pluginmanager.register(p, 'custom_log_plugin')
+
+
+class CustomLogPlugin:
+    """
+    Custom log class to remove log from console
+    and place it inside per-test log files
+    """
+    def __init__(self, config: pytest.Config) -> None:
+        self.config: pytest.Config = config
+        self.log_per_test: str = config.getoption("log_per_test")
+        self.tests = defaultdict(dict)
+        if self.log_per_test not in ["never", "duplicate"]:
+            self.config.option.showlocals = True
+            self.config.option.reportchars = 'a'
+            self.config.option.tbstyle = 'line'
+            self.config.option.showcapture = 'no'
+            self.config.option.capture = 'fd'
+
+    def _write_log(self, test: str, phases: list = None) -> None:
+        """
+        Write log for test, possibly only selected
+        phases can be written.
+        """
+        if not phases:
+            _phases = ["setup", "call", "teardown"]
+        else:
+            _phases = phases
+        tr = self.tests[test]
+        test_name = test.split("::")[-1]
+        test_name = test_name.translate(
+            str.maketrans('":<>|*? [', "---------", "]()"))
+        logpath = os.path.join(
+            os.path.dirname(self.config.option.log_file),
+            'logs',
+            f'{test_name}.log'
+        )
+        with open(logpath, 'a+') as f:
```

Comment:
Should we check if the target directory exists and create it before the open?

Reply:
The directory should exist as we are using existing path, adding dir creation just to be sure anyway.